bloom-works/guides-template
# Set up Accessibility testing

## PR Description/Summary

This change is related to the [Setup Guides](https://github.com/bloom-works/guides-template/issues/5) task. Modeling the [Civic Actions Accessibility playbook guide](https://accessibility.civicactions.com/playbook/automated-testing), we have added automated JavaScript front end testing using [Cypress](https://docs.cypress.io/guides/overview/why-cypress) in combination with [Axe core](https://github.com/dequelabs/axe-core), an accessibility testing engine for websites.

The accessibility tests run in the PR check after the playwright tests are finished. 

### What was changed

- Set up of Cypress and Axe core testing frameworks via Dockerfile
- Updated sh file in GitHub workflow pr check
- README doc updated with developer instructions for running Cypress tests

### How to check change

- Checkout branch and build project
- Make sure the server is running at `http://localhost:8080/`
- Run `npx cypress run` to run and verify accessibility tests in headless mode or Run `npx cypress open` to use Cypress' UI mode
- Creation and configuration of new tests can be implemented in `cypress/e2e/spec.cy.js`
- Test configurations can be found in `cypress.config.js`

### Additional Notes

** Since we don't have full pages yet, some of the components are showing accessibility errors because for example, not every component has an `h1` currently. So, the current tests I have in place in the spec file, `cypress/e2e/spec.cy.js`, are disabling specific accessibility errors. We will re-enable them once we copy this guides repository and start building out pages.

** Cypress automatically creates a video when it's run in headless mode, I have disabled this feature in the `cypress.congif.js` file so it's not created during the GitHub workflow. To re-enable it in your local env, change the value of `video` to `true` or comment it out.

** If you are running Cypress tests in Docker Desktop and on a virtual machine, there are [additional dependencies](https://docs.cypress.io/guides/continuous-integration/introduction#Dependencies) you will need to install depending on what type of environment you have.

### Screenshot of PR check accessibility tests

![Accessibility tests](https://github.com/bloom-works/guides-template/assets/20325483/c6e87a3d-eb9d-4e63-923b-a8d45b3b8177)
